test/draw_test.py

```diff
@@ -1,7 +1,204 @@
+
+        for draw_line in [draw.line, draw.aaline]:
+            colors, surfaces = lines_set_up()
+            for surface in surfaces:
+                for color in colors:
+                    self.assertTrue(line_is_color(surface, color, draw_line))
+
+    def test_line_gaps(self):
+        """|tags: ignore|
+
+        Tests if the line drawn with line_has_gaps() contains any gaps.
+
+        See: #512
+        """
+
+        def line_has_gaps(surface, draw_line):
+            """Returns True if the line drawn on the surface contains gaps.
+            """
+            width = surface.get_width()
+            color = (255, 255, 255)
+
+            draw_line(surface, color, (0, 0), (width - 1, 0))
+
+            colors = [surface.get_at((width, 0)) for width in range(width)]
+
+            return len(colors) == colors.count(color)
+
+        for draw_line in [draw.line, draw.aaline]:
+            _, surfaces = lines_set_up()
+            for surface in surfaces:
+                self.assertTrue(line_has_gaps(surface, draw_line))
+
+    def test_lines_color(self):
+        """|tags: ignore|
+
+        Tests if the lines drawn with lines_are_color() are the correct color.
+        """
+        def lines_are_color(surface, color, draw_lines):
+            """Draws (aa)lines around the border of the given surface and
+            checks if all borders of the surface only contain the given color.
+            """
+            width = surface.get_width()
+            height = surface.get_height()
+            points = [(0, 0), (width - 1, 0), (width - 1, height - 1),
+                      (0, height - 1)]
+
+            draw_lines(surface, color, True, points)
+
+            borders = get_border_values(surface, width, height)
+            return [color in border for border in borders]
+
+        for draw_lines in [draw.lines, draw.aalines]:
+            colors, surfaces = lines_set_up()
+            for surface in surfaces:
+                for color in colors:
+                    in_border = lines_are_color(surface, color, draw_lines)
+                    self.assertTrue(in_border.count(True) == 4)
+
+    def test_lines_gaps(self):
+        """|tags: ignore|
+
+        Tests if the lines drawn with lines_have_gaps() contain any gaps.
+
+        See: #512
+        """
+
+        def lines_have_gaps(surface, draw_lines):
+            """Draws (aa)lines around the border of the given surface and
+            checks if all borders of the surface contain any gaps.
+            """
+            width = surface.get_width()
+            height = surface.get_height()
+            color = (255, 255, 255)
+            points = [(0, 0), (width - 1, 0), (width - 1, height - 1),
+                      (0, height - 1)]
+
+            draw_lines(surface, color, True, points)
+
+            borders = get_border_values(surface, width, height)
+            return [len(border) == border.count(color) for border in borders]
```

Comment:
I suggest to replace `len(border) == border.count(color)` with the pattern

```python
all(c == color for c in border)
```

Reply:
Will also update this, thanks.